### PR TITLE
LPS-73959 adding integration test

### PR DESCRIPTION
**Pull request by linolaoj**

### CODE REVIEW

**Comment by linolaoj** [comment]
ci:retest

**Comment by inacionery** [comment]
ci:retest

**Comment by vicnate5** [comment]
Test failures not related.

**Comment by brandizzi** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: portal-workflow-kaleo-definition-impl/src/main/java/com/liferay/portal/workflow/kaleo/definition/internal/parser/BaseNodeValidator.java [hunk 1]
```diff
@@ -35,11 +38,28 @@ public void validate(Definition definition, T node)
...
 
+	protected void validateNotifications(T node)
+		throws KaleoDefinitionValidationException {
+
+		Set<Notification> notifications = node.getNotifications();
+
+		if (notifications.stream().anyMatch(
+				notification -> Validator.isNull(notification.getTemplate()))) {
+
+			throw new KaleoDefinitionValidationException(
```
Comment (inacionery) [inline portal-workflow-kaleo-definition-impl/src/main/java/com/liferay/portal/workflow/kaleo/definition/internal/parser/BaseNodeValidator.java]: Hey @linolaoj, please create a specific exception for this in [KaleoDefinitionValidationException](https://github.com/liferay/com-liferay-portal-workflow/blob/master/portal-workflow-kaleo-definition-api/src/main/java/com/liferay/portal/workflow/kaleo/definition/exception/KaleoDefinitionValidationException.java), like this [other](https://github.com/liferay/com-liferay-portal-workflow/blob/master/portal-workflow-kaleo-definition-api/src/main/java/com/liferay/portal/workflow/kaleo/definition/exception/KaleoDefinitionValidationException.java#L39)

Comment (linolaoj) [reply portal-workflow-kaleo-definition-impl/src/main/java/com/liferay/portal/workflow/kaleo/definition/internal/parser/BaseNodeValidator.java]: @inacionery , exception added https://github.com/liferay/com-liferay-portal-workflow/pull/232/commits/cfa5d0f3475bf28b539de07276c8df4995249bff

File: portal-workflow-kaleo-definition-api/src/main/java/com/liferay/portal/workflow/kaleo/definition/exception/KaleoDefinitionValidationException.java [hunk 2]
```diff
@@ -36,6 +36,25 @@ public KaleoDefinitionValidationException(Throwable cause) {
 		super(cause);
 	}
 
+	public static class EmptyNotificationTemplate
+		extends KaleoDefinitionValidationException {
+
+		public EmptyNotificationTemplate(String node) {
+			super(
+				String.format(
+					"The %s node has a empty notification template", node));
```
Comment (inacionery) [inline portal-workflow-kaleo-definition-api/src/main/java/com/liferay/portal/workflow/kaleo/definition/exception/KaleoDefinitionValidationException.java]: Hey @linolaoj, thanks for this change, could you please also add this message to [Language.properties](https://github.com/liferay/com-liferay-portal-workflow/blob/master/portal-workflow-lang/src/main/resources/content/Language.properties). Thx!

Comment (linolaoj) [reply portal-workflow-kaleo-definition-api/src/main/java/com/liferay/portal/workflow/kaleo/definition/exception/KaleoDefinitionValidationException.java]: @inacionery , I just added the message on Language.properties (https://github.com/liferay/com-liferay-portal-workflow/pull/232/commits/fd5835b2a35de538fa5d2bf49c3faeb5074fe631). Does it need to be referenced somewhere?

Comment (inacionery) [reply portal-workflow-kaleo-definition-api/src/main/java/com/liferay/portal/workflow/kaleo/definition/exception/KaleoDefinitionValidationException.java]: We use this in Kaleo Design under the master-private branch, but for now it's not necessary.

**Comment by inacionery** [comment]
ci:retest

**Comment by inacionery** [comment]
ci:retest

**Comment by inacionery** [comment]
Hey @vicnate5 there is some issue with CI?

**Comment by vicnate5** [comment]
ci:retest

**Comment by vicnate5** [comment]
@inacionery Looks there is a CI issue. You can see realtime status at https://testray.liferay.com/dashboard. Right now there is are pulls in Queued, but the servers aren't being used. This is affecting everyone.

**Comment by vicnate5** [comment]
ci:retest

**Comment by inacionery** [comment]
Hey @linolaoj, could you please run gradlew baseline and commit the change. thx!

**Comment by linolaoj** [comment]
@inacionery, packageinfo updated.
Thanks!

**Comment by inacionery** [comment]
ci:retest

**Comment by inacionery** [comment]
ci:retest

**Pull request merged**